File: bndtools/pl.caltha.osgi.bndtools.impl/src/pl/caltha/osgi/bndtools/cmd/GreetingCommand.java
package pl.caltha.osgi.bndtools.cmd;

import java.io.PrintStream;
import java.util.StringTokenizer;

import org.apache.felix.shell.Command;

import pl.caltha.osgi.bndtools.api.Greeting;
import aQute.bnd.annotation.component.Component;
import aQute.bnd.annotation.component.Reference;

@Component
public class GreetingCommand implements Command {

    private Greeting greetingSvc;

    @Reference
    public void setGreeting(Greeting greetingSvc) {
        this.greetingSvc = greetingSvc;
    }

    public void execute(String line, PrintStream out, PrintStream err) {
        StringTokenizer tokenizer = new StringTokenizer(line);
        tokenizer.nextToken(); // discard first token

        String name = "";
        if (tokenizer.hasMoreTokens())
            name = tokenizer.nextToken();

        System.out.println(greetingSvc.sayHello(name));
    }
    
    public String getName() {
        return "greet";
    }
    
    public String getShortDescription() {
        return "Example command";
    }
    
    public String getUsage() {
        return "greet <name>";
    }
}

File: bndtools-scala/pl.caltha.osgi.bndtools-scala.cmd/src/pl/caltha/osgi/bndtools/scala/cmd/GreetingCommand.scala
package pl.caltha.osgi.bndtools.scala.cmd

import java.io.PrintStream
import org.apache.felix.shell.Command
import aQute.bnd.annotation.component.Component
import aQute.bnd.annotation.component.Reference
import pl.caltha.osgi.bntools.scala.api.Greeting
import scala.reflect.BeanProperty

@Component
class GreetingCommand extends Command {

  var greetingService: Greeting = null
  
  @Reference 
  def setGreetingService(greeting: Greeting) = { 
    greetingService = greeting 
  }

  def getName() = "greet"

  def getShortDescription() = "Display a greeting"

  def getUsage() = "greet <name>"

  def execute(line: String, out: PrintStream, err: PrintStream) = {
    line.split("\\s+") match {
      case Array(_, name) => println(greetingService.sayHello(name))
      case _ =>
    }
  }
}